tests/test_class_sh_void_ptr_capsule.py

```diff
@@ -3,26 +3,79 @@
 from pybind11_tests import class_sh_void_ptr_capsule as m
 
 
+class Valid:
+
+    capsule_generated = False
```

Comment:
This looks dangerous.
```
def __init__(self):
  self.capsule_generated = False
```
maybe?
Even if the test only calls it once at the moment, I'd add that `__init__`. Otherwise someone could trip over this when working or experimenting with this test later.